src/contracts/core/DelegationManager.sol

```diff
@@ -510,199 +505,178 @@ contract DelegationManager is
+
+            // delegatedShares for staker to place into queueWithdrawal
+            delegatedSharesToWithdraw[i] = WithdrawableShares.wrap(sharesToWithdraw[i]).toDelegatedShares(totalMagnitudes[i]);
+            uint256 principalSharesToWithdraw = Shares.unwrap(
+                delegatedSharesToWithdraw[i].toShares(depositScalingFactors[staker][strategies[i]])
+            );
+
+            // TODO: maybe have a getter to get shares for all strategies, like getDelegatableShares
+            // check sharesToWithdraw is valid
+            // but for inputted strategies
+            uint256 principalSharesWithdrawable = shareManager.stakerStrategyShares(staker, strategies[i]);
+            require(principalSharesToWithdraw <= principalSharesWithdrawable, WithdrawalExeedsMax());
+
             // Similar to `isDelegated` logic
             if (operator != address(0)) {
                 // forgefmt: disable-next-item
-                _decreaseOperatorShares({
+                _decreaseDelegation({
                     operator: operator, 
                     staker: staker, 
                     strategy: strategies[i], 
-                    shares: shares[i]
+                    delegatedShares: DelegatedShares.unwrap(delegatedSharesToWithdraw[i])
                 });
             }
 
             // Remove active shares from EigenPodManager/StrategyManager
-            if (strategies[i] == beaconChainETHStrategy) {
-                /**
-                 * This call will revert if it would reduce the Staker's virtual beacon chain ETH shares below zero.
-                 * This behavior prevents a Staker from queuing a withdrawal which improperly removes excessive
-                 * shares from the operator to whom the staker is delegated.
-                 * It will also revert if the share amount being withdrawn is not a whole Gwei amount.
-                 */
-                eigenPodManager.removeShares(staker, shares[i]);
-            } else {
-                require(
-                    staker == withdrawer || !strategyManager.thirdPartyTransfersForbidden(strategies[i]),
-                    "DelegationManager._removeSharesAndQueueWithdrawal: withdrawer must be same address as staker if thirdPartyTransfersForbidden are set"
-                );
-                // this call will revert if `shares[i]` exceeds the Staker's current shares in `strategies[i]`
-                strategyManager.removeShares(staker, strategies[i], shares[i]);
-            }
-
-            unchecked {
-                ++i;
-            }
+            // EigenPodManager: this call will revert if it would reduce the Staker's virtual beacon chain ETH shares below zero
+            // StrategyManager: this call will revert if `principalSharesToDecrement` exceeds the Staker's current deposit shares in `strategies[i]`
+            shareManager.removeShares(staker, strategies[i], principalSharesToWithdraw);
```

Comment:
to be clear, this code should never actually revert here, right? because we check that `principalSharesToWithdraw <= principalSharesWithdrawable` already above

Reply:
This is a good point, we actually are double checking this in both DM and {EP,S}M. Will remove these revert comments here but add in the SM and EPM that the decrements should never actually revert since its already checked here.